sdk/core/Azure.Core/src/TelemetryDetails.cs

```diff
@@ -88,7 +86,19 @@ internal static string GenerateUserAgentString(Assembly clientAssembly, string?
                 version = version.Substring(0, hashSeparator);
             }
             runtimeInformation ??= new RuntimeInformationWrapper();
-            var platformInformation = EscapeProductInformation($"({runtimeInformation.FrameworkDescription}; {runtimeInformation.OSDescription})");
+
+            // RFC 9110 section 5.5 https://www.rfc-editor.org/rfc/rfc9110.txt#section-5.5 does not require any specific encoding : "Fields needing a greater range of characters
+            // can use an encoding, such as the one defined in RFC8187." RFC8187 is targeted at parameter values, almost always filename, so using url encoding here instead, which is
+            // more widely used. Since user-agent does not usually contain non-ascii, only encode when necessary.
+            // This was added to support operating systems with non-ascii characters in their release names.
+            string osDescription;
+#if NET8_0_OR_GREATER
```

Comment:
Adding this so eventually we will use this, even though we don't have a .net 8 target right now